src/transformers/modeling_flash_attention_utils.py

```diff
@@ -20,9 +20,18 @@
 import torch
 import torch.nn.functional as F
 
+from .integrations.deepspeed import (  # DeepSpeed seq parallelism (aka Ulysses)
+    is_deepspeed_available,
+    is_deepspeed_sp_enabled,
+)
 from .utils import is_flash_attn_2_available, is_flash_attn_greater_or_equal
 
 
+if is_deepspeed_available():
+    from deepspeed.sequence.layer import _SeqAllToAll
```

Comment:
Not sure I enjoy the fact that we rely on the private `_SeqAllToAll`, anything we should worry about there in the future?

Reply:
Nothing to worry about!